### PR TITLE
precommit: isort

### PR DESCRIPTION
**Pull request by Borda**
use Isort as default import formatting
Partial alternative to #4983

This PR defines Isort as pre-commit hook and lets the bot fix it... rabbit

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
This PR brings codebase improvements mainly by enlisting third-party tools for code clean-up and organizing imports.

### 📊 Key Changes
- Enabled `isort` tool in `.pre-commit-config.yaml` to automatically sort the imports.
- Refactoring import statements across multiple files like `detect.py`, `export.py`, `hubconf.py`, and others to be more organized and readable based on the `isort` standard.
- Set `isort` configuration in `setup.cfg`, specifying a line length of 120 characters and a single-line output for multiple imports.
- Reshuffling import statements and minor updates to spaces and blank lines to adhere to cleaner code standards.

### 🎯 Purpose & Impact
- The use of `isort` 🔄 ensures that the imports are consistently formatted, which makes the code easier to read and maintain.
- With better-structured imports, developers can understand dependencies at a glance 🧐.
- This can potentially reduce merge conflicts related to import lines for contributors in the future ✨.
- Streamlined code patterns contribute to a more stable and standardized codebase 🛠️.

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@Borda awesome, I didn't know about isort! I looked at the results. Could we configure isort to create the default behavior shown on their README? Current PR is turning all multi-line imports into one-per-line, but README shows from file import (p1,p2,p3 etc.) utilizing full line width.

### Current behavior (this PR)
Screen Shot 2021-11-04 at 4 39 52 PM

### Desired behavior (https://github.com/PyCQA/isort README)
Screen Shot 2021-11-04 at 4 38 13 PM

File: setup.cfg [hunk 1]
```diff
@@ -43,3 +43,11 @@ ignore =
     F403
     E302
     F541
+
+
+[isort]
```
Comment (Borda) [inline setup.cfg:48]: @glenn-jocher here is the configuration :]
feel free to edit it how you wish and let the bot update the PR

Comment (glenn-jocher) [reply setup.cfg:48]: @Borda awesome!! Thanks I'll experiment with it.

**Comment by glenn-jocher** [comment]
@Borda PR is merged. I reverted most of the isort settings back to default values except for line length 120. Thank you for your contributions to YOLOv5 🚀 and Vision AI ⭐

**Pull request merged**